# Graded assignment 1
Maris van der Velden
#2003885


# Packages

We need the package `tidyverse`

In [3]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [4]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3        V4          V5        V6        V7         
1 0.5195408 0.86931057 0.9037448 0.118839971 0.2266659 0.9332357 0.692967913
2 0.7044246 0.04650977 0.5863805 0.005208574 0.3155486 0.5175686 0.007780464
  V8        V9        V10        V11       V12       V13       V14      
1 0.1172385 0.9553258 0.60745754 0.6247922 0.7339693 0.1224358 0.8329162
2 0.8108655 0.5210718 0.04319063 0.1764449 0.6498933 0.9571483 0.7304156
  V15       V16       V17       V18       V19       V20        
1 0.9346445 0.5030709 0.6219091 0.6590339 0.6360572 0.771643961
2 0.1652227 0.6329305 0.7407339 0.2337958 0.1096914 0.006504701

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [5]:
df2 <- df1 %>%
  gather()

head(df2)

key value     
1 V1  0.51954083
2 V1  0.70442456
3 V2  0.86931057
4 V2  0.04650977
5 V3  0.90374482
6 V3  0.58638045

## Assignment 2

We will generate some data:

In [6]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [7]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [8]:
df5 <- full_join(df3, df4, by='id')

df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook